src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -5631,7 +5631,8 @@ MethodTableBuilder::PlaceNonVirtualMethods()
 #endif // _DEBUG
 
         if (!fCanHaveNonVtableSlots ||
-            it->GetMethodType() == mcInstantiated)
+            it->GetMethodType() == mcInstantiated ||
+            it->GetAsyncMethodKind() != AsyncMethodKind::NotAsync)
```

Comment:
This would force async/task-returning methods to have real slots. This is the "size"  tradeoff.